v2/googlecloud-to-elasticsearch/src/test/java/com/google/cloud/teleport/v2/elasticsearch/templates/SpannerChangeStreamsToElasticsearchTest.java

```diff
@@ -0,0 +1,165 @@
+      new DataChangeRecord(
+          "partitionToken1",
+          Timestamp.now(),
+          "transactionId2",
+          true,
+          "recordSequence2",
+          "testTable",
+          rowType,
+          updateMods,
+          ModType.UPDATE,
+          ValueCaptureType.OLD_AND_NEW_VALUES,
+          1,
+          1,
+          null);
+  private static final DataChangeRecord deleteRow =
+      new DataChangeRecord(
+          "partitionToken1",
+          Timestamp.now(),
+          "transactionId3",
+          true,
+          "recordSequence3",
+          "testTable",
+          rowType,
+          deleteMods,
+          ModType.DELETE,
+          ValueCaptureType.OLD_AND_NEW_VALUES,
+          1,
+          1,
+          null);
+  private static final List<DataChangeRecord> rows =
+      ImmutableList.of(insertRow, updateRow, deleteRow);
+  private List<String> jsonifiedRows =
+      ImmutableList.of(
+          "{\"TableName\":\"testTable\",\"AccountId\":\"AccountId1\",\"LastUpdate\":\"2022-09-27T11:30:00.123456Z\",\"UserId\":\"UserId1\",\"Balance\":1000}",
+          "{\"TableName\":\"testTable\",\"AccountId\":\"AccountId2\",\"LastUpdate\":\"2022-09-27T11:30:00.123456Z\",\"UserId\":\"UserId1\",\"Balance\":3000}",
+          "{\"TableName\":\"testTable\",\"AccountId\":\"AccountId1\",\"LastUpdate\":\"2022-09-27T11:30:00.123456Z\",\"UserId\":\"UserId2\",\"Balance\":500}",
+          "{\"TableName\":\"testTable\",\"AccountId\":\"AccountId2\",\"LastUpdate\":\"2022-09-27T11:30:00.123456Z\",\"UserId\":\"UserId2\",\"Balance\":2000}",
+          "{\"TableName\":\"testTable\",\"AccountId\":\"AccountId1\",\"LastUpdate\":\"2022-09-27T12:30:00.123456Z\",\"UserId\":\"UserId1\",\"Balance\":500}",
+          "{\"TableName\":\"testTable\",\"AccountId\":\"AccountId2\",\"LastUpdate\":\"2022-09-27T12:30:00.123456Z\",\"UserId\":\"UserId1\",\"Balance\":3500}",
+          "{\"TableName\":\"testTable\",\"AccountId\":\"AccountId1\",\"LastUpdate\":\"2022-09-27T12:30:00.123456Z\",\"UserId\":\"UserId2\",\"Balance\":700}",
+          "{\"TableName\":\"testTable\",\"AccountId\":\"AccountId2\",\"LastUpdate\":\"2022-09-27T12:30:00.123456Z\",\"UserId\":\"UserId2\",\"Balance\":1800}",
+          "{\"TableName\":\"testTable\",\"AccountId\":\"AccountId1\",\"UserId\":\"UserId1\",\"IsDelete\":true}",
+          "{\"TableName\":\"testTable\",\"AccountId\":\"AccountId2\",\"UserId\":\"UserId1\",\"IsDelete\":true}",
+          "{\"TableName\":\"testTable\",\"AccountId\":\"AccountId1\",\"UserId\":\"UserId2\",\"IsDelete\":true}",
+          "{\"TableName\":\"testTable\",\"AccountId\":\"AccountId2\",\"UserId\":\"UserId2\",\"IsDelete\":true}");
+
+  @Rule public final transient TestPipeline pipeline = TestPipeline.create();
+  @Rule public ExpectedException exceptionRule = ExpectedException.none();
+
+  /** Test the {@link SpannerChangeStreamsToElasticsearch} pipeline end-to-end. */
+  @Test
+  public void testSpannerChangeStreamsToElasticsearchE2E() {
```

Comment:
This is not really and end-to-end test as it only tests `DataChangeRecordToJsonFn`.
Please at the minimum rename this test case, or make `DataChangeRecordToJsonFn` a standalone class and add a separate test class for that.
(And testing the template itself E2E is hard or impossible in a unit test. It's OK to not have any E2E tests for now. We should move integration tests to github soon. Adding an E2E integration test after that would make sense).

Reply:
I renamed the test case function.. is that ok for now or you'd prefer the second approach of a standalone class for `DataChangeRecordToJsonFn`?

Reply:
Thanks, either works!